### ISSUE TITLE
Swinging on Vine/Rope!

### ISSUE THREAD

**Issue by xEvoGx** [thestonefox/SteamVR_Unity_Toolkit#396]
Perhaps an extension of Climbing (actually pretty certain it should be), but it would be cool to be able to swing similar to **Windlands**. 

Climbing is already pretty satisfying, but I wanted to be able to swing on the rope climb section. :smile: 

Thanks yet again!

**Comment by HyppoNymous** [comment]
You can use some classic rope object made of bones and just make it grabable so you cand grab it and ungrab it.
https://www.youtube.com/watch?v=GaAjve4NZd0

**Comment by xEvoGx** [comment]
Yup already did. The rope just swings regardless of the position at which I _grabbed_. In other words, the grab function works if I grab the rope proper, but beyond that it doesn't then keep the rope steady. Still allowed to climb, but the rope behaves as if I'm not even touching it.

That being said, there's no swinging. Not only would we have to register a grip, but we'd need both the ability to climb the rope, and have the rope's movement translate to the playarea. This is far more complex of an issue.

**Comment by HyppoNymous** [comment]
You need to instantiate some object to simulate your body catching one of the rope collision cylinder , adding forces to the body obecjt will make physic joints swing.
This is complex, instead i would go for some easy way
https://www.youtube.com/watch?v=AnVst7_C250
Using a rigid cylinder with a joint at it's top , this would be the rope. 
When you grab the rope , you instantiate a "body" physic object as child of the rope with a joint at the position you grabbed the rope cyliner. Adding forces to the child object will make it swing.
When you ungrab the rope you destory the child physic object and it's joint.

**Comment by xEvoGx** [comment]
No of course Hyppo...

...and you do realize I'm suggesting all of these to augment the Kit right? I really appreciate your taking the time to suggest ways for myself to make it work, but that isn't necessary. Really, what I've been doing with Stone for some time is take what he's laid out, and playtest and ideally find ways to make the kit even more robust. Climbing was clearly added later on in this kits cycle, so it's only fitting that additional options are explored. No they're not necessary, but if I can even tickle someones brain a bit to consider augmenting the system, why not right?

Cheers man

**Comment by HyppoNymous** [comment]
Right, the more the kit has , the best it is for users.

**Comment by xEvoGx** [comment]
Essentially yup.

**Comment by mattboy64** [comment]
The climbing currently does not allow for moving objects. It's a known bug, and one I'll eventually get to, but if somebody wants to take a crack at it before I have the time, that would be fantastic.

**Comment by thestonefox** [comment]
Good reference https://www.youtube.com/watch?v=hETJ_AgulgY from @fuseman

**Comment by JoshGalvin** [comment]
Hmm, you could make some really insane obstacle courses with grabbable moving objects.
